### PR TITLE
Pv2 1346 update is levy payer flag for levy account and apprenticeships

### PR DESCRIPTION
**Pull request by samuel-dev-dba**

### CODE REVIEW

File: src/SFA.DAS.Payments.DataLocks.Application.UnitTets/Mapping/ApprovalsApprenticeshipEventsMappingTests.cs [hunk 1]
```diff
@@ -217,6 +218,28 @@ public void Maps_DataLockTriageApprovedEvent_To_UpdatedApprenticeshipDataLockTri
                 .All(pe => model.ApprenticeshipPriceEpisodes.Any(ape => ape.StartDate == pe.FromDate && ape.Cost == pe.Cost && ape.EndDate == pe.ToDate))
                 .Should().BeTrue();
         }
+        
+        [Test]
+        public void Maps_IsLevyPayer_From_ApprenticeshipUpdated_To_ApprenticeshipModel_Correctly()
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.DataLocks.Application.UnitTets/Mapping/ApprovalsApprenticeshipEventsMappingTests.cs]: these two tests could be generalised into a single test with the events passed in

File: src/SFA.DAS.Payments.DataLocks.Application/Services/ApprenticeshipProcessor.cs [hunk 2]
```diff
@@ -212,7 +213,35 @@ public async Task ProcessPaymentOrderChange(PaymentOrderChangedEvent paymentOrde
...
         }
-        
+
+        public async Task ProcessApprenticeshipForNonLevyPayerEmployer(long accountId)
+        {
+            try
+            {
+                logger.LogDebug($"Processing the apprenticeship Employer Is Non Levy Payer event for Account id: {accountId}");
+               
+                var updatedApprenticeships = await apprenticeshipService.UpdateApprenticeshipEmployerIsLevyPayerFlag(accountId).ConfigureAwait(false);
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.DataLocks.Application/Services/ApprenticeshipProcessor.cs]: could this be better named something like "GetLatestApprenticeshipEmployerIsLevyPayerFlag"? it isn't really clear from it being named Update that it is in fact calling something else to get it and then update locally

File: src/SFA.DAS.Payments.DataLocks.Domain/Services/Apprenticeships/ApprenticeshipService.cs [hunk 3]
```diff
@@ -66,6 +69,22 @@ public async Task<List<ApprenticeshipDuplicateModel>> NewApprenticeship(Apprenti
...
+        public async Task<List<ApprenticeshipModel>> UpdateApprenticeshipEmployerIsLevyPayerFlag(long accountId,CancellationToken cancellation = default(CancellationToken))
+        {
+            var apprenticeships = await repository.GetEmployerApprenticeships(accountId, cancellation).ConfigureAwait(false);
+
+            if (apprenticeships == null || apprenticeships.Count == 0)
+            {
+                return  new List<ApprenticeshipModel>();
+            }
+
+            apprenticeships.ForEach(x => x.IsLevyPayer = false);
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.DataLocks.Domain/Services/Apprenticeships/ApprenticeshipService.cs]: are they all supposed to be set to false?

File: src/SFA.DAS.Payments.DataLocks.Application/Services/ApprenticeshipProcessor.cs (hunk 2 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.DataLocks.Application/Services/ApprenticeshipProcessor.cs]: Name changed

**Pull request merged**